### PR TITLE
sanitize whole string not only first character

### PR DESCRIPTION
**Pull request by gpomykala**
i have a change for sanitizer, please see the performance impact:

```
benchcmp is deprecated in favor of benchstat: https://pkg.go.dev/golang.org/x/perf/cmd/benchstat
benchmark                                                             old ns/op     new ns/op     delta
BenchmarkSanitize/prepend_tab-24                                      0.00          0.00          -10.53%
BenchmarkSanitize/prepend_carriage_return-24                          0.00          0.00          +0.00%
BenchmarkSanitize/handle_forbidden_char_after_fake_separator_;-24     0.00          0.00          +333.33%
BenchmarkSanitize/handle_forbidden_char_after_fake_separator_,-24     0.00          0.00          +400.00%
BenchmarkSanitize/prepend_=-24                                        0.00          0.00          +69.23%
BenchmarkSanitize/prepend_+-24                                        0.00          0.00          +31.25%
BenchmarkSanitize/prepend_--24                                        0.00          0.00          -20.83%
BenchmarkSanitize/prepend_@-24                                        0.00          0.00          +0.00%
```

old impl (first character only) O(1) time complexity
```
grzegorz@gpom-ubuntu:~/go/src/github.com/skiprco/go-utils/converters/csv$ cat old
goos: linux
goarch: amd64
pkg: github.com/skiprco/go-utils/v4/converters/csv
cpu: AMD Ryzen 9 3900X 12-Core Processor            
BenchmarkSanitize/prepend_tab-24                1000000000               0.0000019 ns/op
BenchmarkSanitize/prepend_carriage_return-24            1000000000               0.0000015 ns/op
BenchmarkSanitize/handle_forbidden_char_after_fake_separator_;-24               1000000000               0.0000006 ns/op
BenchmarkSanitize/handle_forbidden_char_after_fake_separator_,-24               1000000000               0.0000005 ns/op
BenchmarkSanitize/prepend_=-24                                                  1000000000               0.0000013 ns/op
BenchmarkSanitize/prepend_+-24                                                  1000000000               0.0000016 ns/op
BenchmarkSanitize/prepend_--24                                                  1000000000               0.0000024 ns/op
BenchmarkSanitize/prepend_@-24                                                  1000000000               0.0000014 ns/op
PASS
ok      github.com/skiprco/go-utils/v4/converters/csv   0.019s
```

new impl (whole string scanned) O(n) time complexity
```
grzegorz@gpom-ubuntu:~/go/src/github.com/skiprco/go-utils/converters/csv$ cat new
goos: linux
goarch: amd64
pkg: github.com/skiprco/go-utils/v4/converters/csv
cpu: AMD Ryzen 9 3900X 12-Core Processor            
BenchmarkSanitize/prepend_+-24          1000000000               0.0000021 ns/op
BenchmarkSanitize/prepend_--24          1000000000               0.0000019 ns/op
BenchmarkSanitize/prepend_@-24          1000000000               0.0000014 ns/op
BenchmarkSanitize/prepend_tab-24        1000000000               0.0000017 ns/op
BenchmarkSanitize/prepend_carriage_return-24            1000000000               0.0000015 ns/op
BenchmarkSanitize/handle_forbidden_char_after_fake_separator_;-24               1000000000               0.0000026 ns/op
BenchmarkSanitize/handle_forbidden_char_after_fake_separator_,-24               1000000000               0.0000025 ns/op
BenchmarkSanitize/prepend_=-24                                                  1000000000               0.0000022 ns/op
PASS
ok      github.com/skiprco/go-utils/v4/converters/csv   0.018s
```

### CODE REVIEW

File: converters/csv/csv_sanitizer.go [hunk 1]
```diff
@@ -13,15 +11,58 @@ var forbiddenChars = map[rune]interface{}{
...
+
+// we only use comma as a separator in CSV export,
+// no way to know upfront what is being used as separator in particular file
+var csvSeparators = map[rune]interface{}{
+	',': nil,
+	';': nil,
+}
+
 func Sanitize(input string) string {
+	//return sanitizeFirst(input)
```
Comment (JensSkipr) [inline converters/csv/csv_sanitizer.go:24]: I suppose this commented code can be removed

File: converters/csv/csv_sanitizer.go [hunk 2]
```diff
@@ -13,15 +11,58 @@ var forbiddenChars = map[rune]interface{}{
...
 	}
-	first, _ := utf8.DecodeRuneInString(input)
-	if _, ok := forbiddenChars[first]; ok {
-		// ensure that cell does not begin with any of forbidden chars
-		// https://owasp.org/www-community/attacks/CSV_Injection
-		return "_" + input
+
+	runes := []rune(input)
+
+	for idx := len(runes) - 1; idx > -1; idx-- {
```
Comment (JensSkipr) [inline converters/csv/csv_sanitizer.go:45]: Non-blocking: I'm aware most of the time regex is slower, but did you try on this particular case? Would at least simplify the code a lot.

**Review by JensSkipr — APPROVED** [review]

File: converters/csv/csv_sanitizer.go (hunk 2 above)
Comment (gpomykala) [reply converters/csv/csv_sanitizer.go:45]: was thinking of it and it is not really cleaner with regexp. regexp will return indexes of matches at best but it's not straightforward to fetch a character by index in Go, still some rune arithmetics is needed to get n+1 character and we still need to insert characters into a string which is not natively supported by Go neither its standard lib

File: converters/csv/csv_sanitizer.go (hunk 1 above)
Comment (gpomykala) [reply converters/csv/csv_sanitizer.go]: ok

File: converters/csv/csv_sanitizer.go (hunk 2 above)
Comment (JensSkipr) [reply converters/csv/csv_sanitizer.go:44]: FYI, I meant using e.g. https://pkg.go.dev/regexp#Regexp.ReplaceAllStringFunc. This way the regex func will handle the replace for you, no need for indexes. But probably slower than current implementation anyway.

**Pull request merged**